infobar-with-cta/client.js

```diff
@@ -43,8 +45,8 @@
         Evergage.cashDom("body").css({ "margin-top": "0", "margin-bottom": "0" });
     }
 
-    function control() {
-
+    function control(context) {
+        return !!context.contentZone;
```

Comment:
Control function can only return a Promise or void, and can't return a boolean. Just do what you did in apply()
```suggestion
        if (!context.contentZone) return;
```

Reply:
Hmmmm where do you see that? I thought I had tested this before and it worked as expected 🤔 

I think I see what you're talking about when I check out the type definition for `control` [here](https://github.com/evergage/evergage-product/blob/e2c06a5ac7a207db6846b6bd42d5e6aa2e6e101f/analytics/excavator/src/main/resources/built-in/build/handlebars-templates/src/services/TemplateService.ts#L120), though I'm not sure that returning void would work here then?

Reply:
https://github.com/evergage/evergage-product/blob/e2c06a5ac7a207db6846b6bd42d5e6aa2e6e101f/analytics/excavator/src/main/resources/built-in/build/handlebars-templates/src/Template.ts

Reply:
I just tested this out and neither the current code or your suggestion prevented the control stat from being sent when a content zone isn't selected; it seems like we actually need to do something like this:
```
return new Promise(resolve => { if (!!context.contentZone) resolve(!!context.contentZone); });
```

Reply:
I looked at TemplateService.ts more closely, and actually, the reason why neither your original solution nor my suggested change works to block the control impression is due to the fact that void lands in the `else` statement, firing the control impression in the downstream anyway.
<img width="1287" alt="Screen Shot 2021-01-05 at 5 37 31 PM" src="https://user-images.githubusercontent.com/32201252/103719223-66416480-4f7d-11eb-88e6-f1fe73e1fb92.png">

I tried your new solution with the hanging Promise and that does seem to work 👍  I would simplify to the following:
`return new Promise(resolve => { if (context.contentZone) resolve(true) });`